src/Nethermind/Nethermind.BeaconNode.Test/Storage/MemoryStoreProviderTest.cs

```diff
@@ -0,0 +1,156 @@
+            Console.WriteLine("");
+
+            Console.WriteLine("Genesis state last block root: {0}, latest block header parent {1}, signing root {2}",
+                genesisState!.BlockRoots.Last(), genesisState!.LatestBlockHeader.ParentRoot, genesisState!.LatestBlockHeader.SigningRoot());
+            Console.WriteLine("State 1 last block root: {0}, latest block header parent {1}, signing root {2}",
+                block1State!.BlockRoots.Last(), block1State!.LatestBlockHeader.ParentRoot, block1State!.LatestBlockHeader.SigningRoot());
+            Console.WriteLine("State 2 last block root: {0}, latest block header parent {1}, signing root {2}",
+                block2State!.BlockRoots.Last(), block2State!.LatestBlockHeader.ParentRoot, block2State!.LatestBlockHeader.SigningRoot());
+            Console.WriteLine("");
+
+            Console.WriteLine("Genesis state last state root: {0}, last historical root {1}",
+                genesisState!.StateRoots.LastOrDefault(), genesisState!.HistoricalRoots.LastOrDefault());
+            Console.WriteLine("State 1 last state root: {0}, last historical root {1}",
+                block1State!.StateRoots.Last(), block1State!.HistoricalRoots.LastOrDefault());
+            Console.WriteLine("State 2 state last state root: {0}, last historical root {1}",
+                block2State!.StateRoots.Last(), block2State!.HistoricalRoots.LastOrDefault());
+            Console.WriteLine("Head state last state root: {0}, last historical root {1}",
+                headState!.StateRoots.Last(), headState!.HistoricalRoots.LastOrDefault());
+            Console.WriteLine("");
+
+            BeaconBlockHeader state1LatestHeader = block1State!.LatestBlockHeader;
+            Console.WriteLine(
+                "State 1 latest header, slot: {0}, parent {1}, body root {2}, state root {3}, signature {4}, signing root {5}",
+                state1LatestHeader.Slot, state1LatestHeader.ParentRoot, state1LatestHeader.BodyRoot,
+                state1LatestHeader.StateRoot, state1LatestHeader.Signature,
+                state1LatestHeader.SigningRoot());
+            Console.WriteLine(
+                "-- compare to Block 1, slot: {0}, parent {1}, body root {2}, state root {3}, signature {4}, signing root {5}",
+                block1!.Slot, block1!.ParentRoot, block1.Body.HashTreeRoot(miscellaneousParameters, maxOperationsPerBlock),
+                block1!.StateRoot, block1!.Signature,
+                block1!.SigningRoot(miscellaneousParameters, maxOperationsPerBlock));
+
+            // Assert
+            block2!.ParentRoot.ShouldBe(block1Root);
+
+            Hash32 block1SigningRoot = block1!.SigningRoot(miscellaneousParameters, maxOperationsPerBlock);
+            Hash32 block1HashTreeRoot = block1!.HashTreeRoot(miscellaneousParameters, maxOperationsPerBlock);
+            
+            block1SigningRoot.ShouldBe(block1Root);
+            
+            block1.ParentRoot.ToString().ShouldStartWith("0x7f4520eb");
```

Comment:
good if you add a string description for each assert if you have three asserts in a row - this will allow in the future to know exactly which one fails (in NUnit it is jus a simple third string parameter - not sure about Should